controllers/release_controller.go

```diff
@@ -127,8 +137,41 @@ func (r *ReleaseReconciler) getReleaseStrategy(ctx context.Context, releaseLink
 	return releaseStrategy, nil
 }
 
+// getTargetReleaseLink gets a ReleaseLink by following the spec.target field in a given ReleaseLink and returning
+// it only if it targets the passed ReleaseLink and they both reference the same Application.
+func (r *ReleaseReconciler) getTargetReleaseLink(ctx context.Context, releaseLink *v1alpha1.ReleaseLink) (*v1alpha1.ReleaseLink, error) {
+	releaseLinks := &v1alpha1.ReleaseLinkList{}
+	opts := []client.ListOption{
+		client.InNamespace(releaseLink.Spec.Target),
+		client.MatchingFields{"spec.target": releaseLink.Namespace},
+	}
+
+	if err := r.List(ctx, releaseLinks, opts...); err != nil {
+		return nil, err
+	}
+	if len(releaseLinks.Items) == 0 {
```

Comment:
Let's remove this and improve the comment in line 161. The check is already covered by the loop.